src/main/java/backend/resource/TurboLabel.java

```diff
@@ -4,153 +4,185 @@
 
-    private final String actualName;
     private final String colour;
-
-    private void ______TRANSIENT_FIELDS______() {
-    }
-
     private final String repoId;
 
-    private void ______CONSTRUCTORS______() {
-    }
-
-    public TurboLabel(String repoId, String name) {
-        this.actualName = name;
-        this.colour = "ffffff";
-        this.repoId = repoId;
-    }
 
     public TurboLabel(String repoId, String colour, String name) {
-        this.actualName = name;
+        this.fullName = name;
+
+        String[] splitted = splitKeyword(name);
+        this.groupName = splitted[0];
+        this.grouping = determineGrouping(splitted[1]);
+        this.shortName = splitted[2];
+
         this.colour = colour;
         this.repoId = repoId;
     }
 
-    public static TurboLabel nonexclusive(String repoId, String group, String name) {
-        return new TurboLabel(repoId, joinWith(group, name, false));
-    }
-
-    public static TurboLabel exclusive(String repoId, String group, String name) {
-        return new TurboLabel(repoId, joinWith(group, name, true));
+    public TurboLabel(String repoId, String name) {
+        this(repoId, DEFAULT_COLOR, name);
     }
 
     /**
      * Copy constructor
      */
     public TurboLabel(TurboLabel label) {
-        this.actualName = label.getActualName();
-        this.colour = label.getColour();
-        this.repoId = label.getRepoId();
+        this(label.getRepoId(), label.getColour(), label.getFullName());
     }
 
     public TurboLabel(String repoId, Label label) {
-        this.actualName = label.getName();
-        this.colour = label.getColor();
-        this.repoId = repoId;
+        this(repoId, label.getColor(), label.getName());
     }
 
     public TurboLabel(String repoId, SerializableLabel label) {
-        this.actualName = label.getActualName();
-        this.colour = label.getColour();
-        this.repoId = repoId;
+        this(repoId, label.getColour(), label.getFullName());
     }
 
-    private void ______METHODS______() {
+
+    /**
+     * Extracts group name, group delimiter and short name from a keyword
+     * group name is name of the group associated with a keyword
+     * short name is name of the keyword after omitting its group name 
+     * Example: "priority.high" -> group name = "priority", group delimiter = ".", short name = "high"
+     * @param keyword
+     * @return String array of group name, group delimiter and short name
+     */
+    private static String[] splitKeyword(String keyword) {
+        Pattern p = Pattern.compile(String.format("^([^\\%s\\%s]+)(\\%s|\\%s)([^\\%s\\%s]*)",
+            EXCLUSIVE_DELIMITER, NONEXCLUSIVE_DELIMITER,
+            EXCLUSIVE_DELIMITER, NONEXCLUSIVE_DELIMITER,
+            EXCLUSIVE_DELIMITER, NONEXCLUSIVE_DELIMITER));
+        Matcher m = p.matcher(keyword);
+
+        if (m.find()) return new String[] {m.group(1), m.group(2), m.group(3)};
+
+        return new String[] {"", "", keyword};
```

Comment:
This reads better (i.e., use guard clauses to deal with the 'abnormal' cases, not the 'normal' case).
```java
if (!m.find()) return new String[] {"", "", keyword};

return new String[] {m.group(1), m.group(2), m.group(3)};
```